domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java

```diff
@@ -906,46 +904,25 @@ static String durationMsg(String descriptionKey, String sql, long startNanoTime)
   }
 
   /**
-   * Run a parameterized/prepared statement and
-   * run it with a list of sets of parameters.
+   * Run a parameterized/prepared statement with a list of tuples as parameters.
+   * This is atomic, if one Tuple fails the complete list fails: all or nothing.
    *
    * @param sql - the SQL command to run
-   * @param params - there is one list entry for each SQL invocation containing the parameters
-   *                    {@code $} placeholders.
-   * @return the reply from the database
+   * @param params - there is one list entry for each SQL invocation containing the
+   *                    parameters for the {@code $} placeholders.
+   * @return the reply from the database, one RowSet per params Tuple. null if params.size() == 0.
    */
-  public Future<List<RowSet<Row>>> execute(String sql, List<Tuple> params) {
+  public Future<RowSet<Row>> execute(String sql, List<Tuple> params) {
 
     try {
+      if (params.size() == 0) {
+        return Future.succeededFuture();
+      }
```

Comment:
Do nothing? Or should it fall back to `execute(String sql)`?

Reply:
"params - there is one list entry for each SQL invocation"

`sql` is run two times if params.size() == 2.
`sql` is run one time if params.size() == 1.
`sql` is run zero times if params.size() == 0.

I've added javadoc and unit tests for these cases.

Reply:
Thanks for updating the javadoc to make it explicit.